network/packet-filtering-firewalls.md

```diff
@@ -110,3 +169,17 @@ $ firewall-cmd --reload
 ```
 
 Restarting `dockerd` daemon inserts the interface into the `docker` zone.
+
+## Docker and ufw
+
+[Uncomplicated Firewall](https://launchpad.net/ufw){: target="_blank" rel="noopener"}
+(ufw) is a frontend that ships with Debian and Ubuntu,
+and it lets you manage firewall rules. Docker and ufw use iptables in ways
+that make them incompatible with each other.
+
+When you publish a container's ports using Docker, traffic to and from that
+container gets diverted before it goes through the ufw firewall settings.
+Docker routes container traffic in the `nat` table, which means that packets
+are diverted before it reaches the `filter` table that ufw uses. Packets are
+routed before the firewall rules can be applied, effectively ignoring your
+firewall configuration.
```

Comment:
I think this advice also applies to firewalld? (_differs a little bit with it's zones_)

A UFW specific issue that differs behaviour with Docker is this: https://github.com/moby/moby/issues/27817